splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -1692,4 +1693,50 @@ public void addMinRetentionPeriodColumn(TransactionController tc) throws Standar
                     SYSTABLESRowFactory.MIN_RETENTION_PERIOD));
         }
     }
+
+    public void setJavaClassNameColumnInSysAliases(TransactionController tc) throws StandardException {
+        SchemaDescriptor sd = getSystemSchemaDescriptor();
+        TabInfoImpl ti=getNonCoreTI(SYSALIASES_CATALOG_NUM);
+        faultInTabInfo(ti);
+
+        FormatableBitSet columnToReadSet=new FormatableBitSet(SYSALIASESRowFactory.SYSALIASES_COLUMN_COUNT);
+        FormatableBitSet columnToUpdateSet=new FormatableBitSet(SYSALIASESRowFactory.SYSALIASES_COLUMN_COUNT);
+        for(int i=0;i<SYSALIASESRowFactory.SYSALIASES_COLUMN_COUNT;i++){
+            // we do not want to read the saved serialized plan
+            if (i+1 == SYSALIASESRowFactory.SYSALIASES_JAVACLASSNAME) {
```

Comment:
From DB-9388, I learnt that for rows in dictionary table, partial row update does not seem to work well. To be safe, we should read all the columns of the row, and also provide the complete update row for update. That is, set all the bits for both `columnToReadSet` and `columnToUpdateSet`.

Reply:
Thanks for the information, I did not know that partial row updates are not working properly, we should investigate this further and fix it.
I will follow your proposed safe measure and mark all columns for read and update.